class.jetpack.php

```diff
@@ -4253,15 +4253,21 @@ function build_connect_url( $raw = false, $redirect = false, $from = false, $reg
 			}
 		} else {
 
-			// Checking existing token
-			$response = Jetpack_Client::wpcom_json_api_request_as_blog(
-				sprintf( '/sites/%d', $site_id ) .'?force=wpcom',
-				'1.1'
-			);
+			// Let's check the existing blog token to see if we need to re-register. We only check once per minute
```

Comment:
I believe all this logic should be encapsulated into a function, as part of the condition for entering the first part of the if-statement. That way no recursion is necessary:

```
if ( $register || ! $token || ! $site_id || ! $this->check_tokens_are_valid() ) {
  // render register URL
} else {
  // render authorize URL
}
```

In fact, if this works out you may be able to drop the $register arg completely from the function.

Reply:
Thanks for the review. I agree that we should probably refactor a bit. I'm not sure that we can get rid of the raw option which ensure we only call out once, but it could at least make things a bit more readable. 👍